Sources/AsyncAlgorithms/Locking.swift

```diff
@@ -17,24 +17,35 @@ import Glibc
 import WinSDK
 #endif
 
-internal struct Lock {
+@usableFromInline
+internal class Lock {
```

Comment:
umm why is this a non-final class? If we are moving to a class then we should just use `Mutex` from Synchronization

Reply:
I removed all of the changes here since the latest code now uses `Mutex`.